### ISSUE TITLE
vSphere volume doesn't exist after pod restarting

### ISSUE THREAD

**Issue by alex97545** [aws/eks-anywhere#970]
vSphere volume doesn't exist after pod restarting. vSphere volume is created (in the Rancher) without any issues but after restarting the pod there is an error (NFS volume is working fine after restarting a pod):

MountVolume.SetUp failed for volume "rgn-pv" : mount failed: exit status 32 Mounting command: mount Mounting arguments: -o bind /var/lib/kubelet/plugins/kubernetes.io/vsphere-volume/mounts/[SitCenter 55] PV_TEST /var/lib/kubelet/pods//volumes/kubernetes.io~vsphere-volume/rgn-pv Output: mount: /var/lib/kubelet/pods//volumes/kubernetes.io~vsphere-volume/rgn-pv: special device /var/lib/kubelet/plugins/kubernetes.io/vsphere-volume/mounts/[SitCenter 55] PV_TEST does not exist.

Should I add vSphere as a Cloud Provider to the coniguration or something else?

**Environment**:
- EKS Anywhere
- Rancher v2.6.3
- vSphere

**Comment by vignesh-goutham** [comment]
Could you explain the environment a bit more. How is Rancher involved in here? We use vsphere cloud provider to help us out with volumes. EKS-A would deploy the cloud provider when you create a cluster. 

Can you get a kubectl get pods -A, to check if there are any issues over there?

**Comment by alex97545** [comment]
So we have onprem storage on the VSphere and we need to connect this storage to the pod.

Output of kubectl get pods -A:

sitcenter                           ict-db-0                                                         0/1     ContainerCreating   0          40m

There are some errors in the PVC logs:

<html>
<body>


Reason | Updated | Message
-- | -- | --
ExternalProvisioning (6) | 1 mins ago | waiting for a volume to be created, either by external provisioner "csi.vsphere.vmware.com" or manually created by system administrator
Provisioning (7) | 1.3 mins ago | External provisioner is provisioning volume for claim "sitcenter/pv-test"
ProvisioningFailed (7) | 1.3 mins ago | failed to provision volume with StorageClass "standard": rpc error: code = Internal desc = failed to create volume. Error: ServerFaultCode: Required parameter entities is missing while parsing call information for method HasUserPrivilegeOnEntities at line 2, column 66 while parsing SOAP body at line 2, column 60 while parsing SOAP envelope at line 2, column 0 while parsing HTTP request for method hasUserPrivilegeOnEntities on object of type vim.AuthorizationManager at line 1, column 0


</body>
</html>

And pod logs:
<html>
<body>


Reason | Updated | Message
-- | -- | --
FailedMount (8) | 55 secs ago | (combined from similar events): MountVolume.SetUp failed for volume "pv-test" : mount failed: exit status 32 Mounting command: mount Mounting arguments: -o bind /var/lib/kubelet/plugins/kubernetes.io/vsphere-volume/mounts/vmfs/volumes/60d555ef-0e93a498-4981-704d7bc234ca/test-folder/pv-test.vmdk /var/lib/kubelet/pods/7f1f0480-66ff-48f8-a31b-00d914068232/volumes/kubernetes.io~vsphere-volume/pv-test Output: mount: /var/lib/kubelet/pods/7f1f0480-66ff-48f8-a31b-00d914068232/volumes/kubernetes.io~vsphere-volume/pv-test: special device /var/lib/kubelet/plugins/kubernetes.io/vsphere-volume/mounts/vmfs/volumes/60d555ef-0e93a498-4981-704d7bc234ca/test-folder/pv-test.vmdk does not exist.
FailedAttachVolume (42) | 2.1 mins ago | AttachVolume.NewAttacher failed for volume "pv-test" : cloud provider not initialized properly


</body>
</html>


I tried to set Provisioner as VSphere and csi.vsphere.vmware.com in the StorageClasses and there is no positive results.
I've set Cloud Provider as a vcsphere.
